experimental/packages/opentelemetry-instrumentation/src/instrumentation.ts

```diff
@@ -173,4 +175,29 @@ export abstract class InstrumentationAbstract implements Instrumentation {
     | InstrumentationModuleDefinition
     | InstrumentationModuleDefinition[]
     | void;
+
+  /**
+   * Execute instrumentation event hook, if configured, and log any errors.
+   * Any semantics of the event type and eventInfo are defined by the specific instrumentation.
+   * @param hookHandler The optional hook handler which the user has configured via instrumentation config\
+   * @param eventName The name of the event being triggered, for logging purposes
+   * @param span The span to which the hook should be applied
+   * @param eventInfo The event info to be passed to the hook
+   */
+  protected runInstrumentationEventHook<EventInfoType>(
```

Comment:
This function uses `this._diag` to silently log any errors from hooks. thus the function needs to have access to `this` and is defined as a class method.

It is marked `protected` so that we can only run it from a class method but it is not exposed on the public API, similarly to the existing `_wrap`, `_unwrap` etc functions.